client/log_verifier.go

```diff
@@ -28,20 +29,32 @@ import (
 
 // logVerifier contains state needed to verify output from Trillian Logs.
 type logVerifier struct {
-	hasher hashers.LogHasher
-	pubKey crypto.PublicKey
-	v      merkle.LogVerifier
+	hasher           hashers.LogHasher
+	pubKey           crypto.PublicKey
+	v                merkle.LogVerifier
+	LeafIdentityAlgo LeafIdentity
 }
 
 // NewLogVerifier returns an object that can verify output from Trillian Logs.
 func NewLogVerifier(hasher hashers.LogHasher, pubKey crypto.PublicKey) LogVerifier {
 	return &logVerifier{
-		hasher: hasher,
-		pubKey: pubKey,
-		v:      merkle.NewLogVerifier(hasher),
+		hasher:           hasher,
+		pubKey:           pubKey,
+		v:                merkle.NewLogVerifier(hasher),
+		LeafIdentityAlgo: Sha256,
 	}
 }
 
+// LeafIdentity computes the identity value for a leaf.
+type LeafIdentity func(leaf []byte) ([]byte, error)
+
+// Sha256 computes the sha256 of the input.
+func Sha256(data []byte) ([]byte, error) {
```

Comment:
You don't even know that a personality is using SHA256 at this stage so this is still dangerous :)

But, I believe that `TrillianLogServer` will set `LeafIdentityHash` to `H(leaf_value)` on your behalf (with the correct hasher for the tree) iff you don't fill in an explicit `LeafIdentityHash` - does that not work for your usecase?